src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -139,15 +140,18 @@ public Category findCategory(ArrayList<Category> monthCategoryData, String name)
                 return c;}}
         throw new NoSuchElementException();}
     /**
-     * Sets the information needed to create a CategoryCreatorInputData to call the createMonthObject method in CategoryFactory
-     * @return CategoryCreatorInputData MonthObjectFactoryInputData Object specifically used in CategoryFactory for the createMonthObject method.
+     * Sets information needed to create a CategoryCreatorInputData to call createMonthObject method in CategoryFactory
+     * @return CategoryCreatorInputData MonthObjectFactoryInputData Object specifically used in CategoryFactory
+     * for the createMonthObject method.
      */
-    private CategoryCreatorInputData setCategoryCreatorInfo(){
-        return CategoryCreatorInputData categoryCreatorInputData = new CategoryCreatorInputData(categoryID.getName(), valueDouble);}
+    private MonthObjectFactoryInputData setCategoryCreatorInfo(CategoryID categoryIDAdd){
+        return new CategoryCreatorInputData(categoryIDAdd.getName(), toDouble(categoryIDAdd.getValue()));}
```

Comment:
You need to change these to accommodate for the setters instead of the constructor (so creating the object and then calling setName and setBudget).